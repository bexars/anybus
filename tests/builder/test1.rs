use anybus::{BusRiderRpc, bus_uuid};
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[bus_uuid("123e4567-e89b-12d3-a456-426614174000")]
pub struct NumberMessage {
    pub value: u32,
}

impl BusRiderRpc for NumberMessage {
    type Response = NumberMessage;
}
// Test to make sure that the builder doesn't build in certain configurations
fn main() {
    use anybus::AnyBusBuilder;

    let bus = AnyBusBuilder::new().build();
    bus.run();
    let handle = bus.handle().clone();
    handle
        .listener()
        .broadcast()
        .rpc()
        .register::<NumberMessage>();
    handle
        .listener()
        .rpc()
        .broadcast()
        .register::<NumberMessage>();
    handle
        .listener()
        .anycast()
        .rpc()
        .register::<NumberMessage>();
    handle
        .listener()
        .rpc()
        .anycast()
        .register::<NumberMessage>();
}
