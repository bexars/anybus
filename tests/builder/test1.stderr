error[E0599]: no method named `rpc` found for struct `anybus::handle::RegistrationBuilder<anybus::handle::NoEndpointId, anybus::handle::CastSet, anybus::handle::NoRpc>` in the current scope
  --> tests/builder/test1.rs:23:10
   |
20 | /     handle
21 | |         .listener()
22 | |         .broadcast()
23 | |         .rpc()
   | |         -^^^ method not found in `anybus::handle::RegistrationBuilder<anybus::handle::NoEndpointId, anybus::handle::CastSet, anybus::handle::NoRpc>`
   | |_________|
   |
   |
   = note: the method was found for
           - `anybus::handle::RegistrationBuilder<EP, anybus::handle::NoCast, anybus::handle::NoRpc>`

error[E0599]: no method named `broadcast` found for struct `anybus::handle::RegistrationBuilder<anybus::handle::NoEndpointId, anybus::handle::NoCast, anybus::handle::RpcSet>` in the current scope
  --> tests/builder/test1.rs:28:10
   |
25 | /     handle
26 | |         .listener()
27 | |         .rpc()
28 | |         .broadcast()
   | |         -^^^^^^^^^ method not found in `anybus::handle::RegistrationBuilder<anybus::handle::NoEndpointId, anybus::handle::NoCast, anybus::handle::RpcSet>`
   | |_________|
   |
   |
   = note: the method was found for
           - `anybus::handle::RegistrationBuilder<EP, anybus::handle::NoCast, anybus::handle::NoRpc>`

error[E0599]: no method named `rpc` found for struct `anybus::handle::RegistrationBuilder<anybus::handle::NoEndpointId, anybus::handle::CastSet, anybus::handle::NoRpc>` in the current scope
  --> tests/builder/test1.rs:33:10
   |
30 | /     handle
31 | |         .listener()
32 | |         .anycast()
33 | |         .rpc()
   | |         -^^^ method not found in `anybus::handle::RegistrationBuilder<anybus::handle::NoEndpointId, anybus::handle::CastSet, anybus::handle::NoRpc>`
   | |_________|
   |
   |
   = note: the method was found for
           - `anybus::handle::RegistrationBuilder<EP, anybus::handle::NoCast, anybus::handle::NoRpc>`
